### PR TITLE
Add pie chart display and histogram display.

### LINKED ISSUE #44: Range results display: bins based on range of possible input
Consider using the max/min possible input, rather than of the max/min actual inputs to generate a histogram.

### LINKED ISSUE #60: make histogram for range results

### PR DESCRIPTION
**Pull request by kevinhquach**
Uses D3.js for the charts and jQuery for graceful degradation.
Closes #44, #60 and #61.
Based on #80.

Note that a front-end testing tool is required to do automated testing for these changes.

### CODE REVIEW

File: src/main/java/com/based/basedsurvey/controller/FillSurveyController.java [hunk 1]
```diff
@@ -121,8 +123,10 @@ private void addResponse(Question q, String value) {
      * @return the HTML inputs for the question
      */
     private String getMultipleChoiceInputs(MultiplechoiceQuestion q) {
-        String s = "";
+        @Language("html")
+        String s = "<hr>";
         s += "<h3> " + q.getPrompt() + " </h3>";
+        if (!Objects.equals(q.getAdditionalInfo(), "")) s += "additional notes: " + q.getAdditionalInfo() + "<br>";
```
[the remark this answers is not in the document]
Comment (Shashaank5) [reply src/main/java/com/based/basedsurvey/controller/FillSurveyController.java:129]: same in the examples below

File: src/main/resources/static/js/results.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                    .attr("transform", function(d) { return `translate(${x(d.x0)}, ${y(d.length)})`})
+                    .attr("width", function(d) { return x(d.x1) - x(d.x0) -1 ; })
+                    .attr("height", function(d) { return height - y(d.length); })
+                    .style("fill", "#69b3a2")
+
+            }
+
+
+            // Initialize with 7 bins
+            update(7)
```
Comment (Shashaank5) [inline src/main/resources/static/js/results.js:141]: Is there any reason for 7 (this is fine if it works)

**Review by DavidHaighton — APPROVED** [review]
```
                                                                                               
    _/        _/_/_/_/    _/_/    _/      _/  _/      _/  _/      _/  _/      _/  _/      _/   
   _/        _/        _/    _/  _/_/    _/  _/_/    _/  _/_/    _/  _/_/    _/  _/_/    _/    
  _/        _/_/_/    _/_/_/_/  _/  _/  _/  _/  _/  _/  _/  _/  _/  _/  _/  _/  _/  _/  _/     
 _/        _/        _/    _/  _/    _/_/  _/    _/_/  _/    _/_/  _/    _/_/  _/    _/_/      
_/_/_/_/  _/_/_/_/  _/    _/  _/      _/  _/      _/  _/      _/  _/      _/  _/      _/      
```

File: src/main/resources/static/js/results.js (hunk 2 above)
Comment (kevinhquach) [reply src/main/resources/static/js/results.js:141]: nah, just an arbitrary default value

**Pull request merged**